File: main/03_adding_data.py
## Adding a single column
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
city = ['Bonn', 'London', 'Toronto', 'Quito', 'Reykjavik'] # new column to be added
df['City'] = city # add city as new column

df = df.assign(City = city) # assign city as new column

df.insert(4, 'City', city) # insert city as new column to position 4 # note: happens inplace


## Adding a single row
# pandas
import pandas as pd
df = pd.read_csv('00_py_data_set.tsv', sep='\t')
dict = {'Name': 'Fabulous', 'Age': 71, 'Random': 'Yes', 'Interesting': True} # dictionary of new row
df = df.append(dict, ignore_index=True) # append dict to df # also works for appending another dataframe

df.loc[5] = dict # add dict as new row # necessary to know, which index is the first 'empty' one

File: main/03_adding_data.jl
## Adding a single column
# DataFrames, CSV
using DataFrames, CSV
df = CSV.read("main/00_jl_data_set.tsv")
city = ["Bonn", "London", "Toronto", "Quito", "Reykjavik"]
insertcols!(df, 5, city=city)


## Adding a single row
# DataFrames, CSV
using DataFrames, CSV
df = CSV.read("main/00_jl_data_set.tsv")
df2 = DataFrame(Name = ["Fabulous"], Age = [71], Random = ["Yes"], Interesting = [true])
df = [df;df2]
